subprojects/shipkit/src/main/groovy/org/shipkit/internal/gradle/util/BranchUtils.java

```diff
@@ -0,0 +1,35 @@
+package org.shipkit.internal.gradle.util;
+
+import org.gradle.api.logging.Logger;
+import org.gradle.api.logging.Logging;
+import org.json.simple.JsonObject;
+import org.shipkit.internal.gradle.git.OpenPullRequest;
+
+public class BranchUtils {
+
+    private static final Logger LOG = Logging.getLogger(BranchUtils.class);
+
+    public static String getHeadBranch(String forkRepositoryName, String headBranch) {
+        return getUserOfForkRepo(forkRepositoryName) + ":" + headBranch;
+    }
+
+    private static String getUserOfForkRepo(String forkRepositoryName) {
+        return forkRepositoryName.split("/")[0];
+    }
+
+    public static OpenPullRequest getOpenPullRequest(JsonObject pullRequest, String filter) {
```

Comment:
I think we can make it simpler. It's not intuitive that "BranchUtils" class holds logic that transforms JSON to PullRequest object. Also, null handling requires me to navigate various methods to understand how it works.

Let's drop this method. We can have a separate class/factory method that is responsible for translating JSON to PullRequest. The client can call "matches" method on ```pullRequest.getRef()``` if branch filtering is needed.

Reply:
Good Point!